tests/Integration/Behaviour/Features/Context/ShopFeatureContext.php

```diff
@@ -49,19 +53,138 @@ public function shopWithNameExists(string $reference, string $shopName)
         SharedStorage::getStorage()->set($reference, new Shop($shopId));
     }
 
+    /**
+     * @Given /I add a shop group "(.+)" with name "(.+)"$/
+     *
+     * @param string $reference
+     * @param string $groupName
+     */
+    public function addShopGroup(string $reference, string $groupName): void
+    {
+        $shopGroup = new ShopGroup();
+        $shopGroup->name = $groupName;
+        $shopGroup->active = true;
+        if (!$shopGroup->add()) {
+            throw new RuntimeException(sprintf('Could not create shop group: %s', Db::getInstance()->getMsgError()));
+        }
+
+        SharedStorage::getStorage()->set($reference, $shopGroup);
+    }
+
+    /**
+     * @Given /I add a shop "(.+)" with name "(.+)" for the group "(.+)"$/
+     *
+     * @param string $reference
+     * @param string $shopName
+     * @param string $shopGroupName
+     */
+    public function addShop(string $reference, string $shopName, string $shopGroupName): void
```

Comment:
Same here, I added exactly the same method in #21593 but with color

Reply:
I think this one will have priority anyway since its target is 177x and yours is develop

Reply:
it makes sense 😭

Reply:
@matthieu-rolland Don't worry. I will rebase your PR ;)

Reply:
ah it's ok, I'll do it 👍  i'll have to do some modifications anyway